app/controllers/concerns/accepts_jwt.rb

```diff
@@ -0,0 +1,12 @@
+module AcceptsJwt
+  extend ActiveSupport::Concern
+
+  def find_or_create_jwt(payload = nil)
+    @find_or_create_jwt ||=
+      if payload
+        Jwt.create!(jwt_payload: payload).tap { |j| session[:jwt_id] = j.id }
+      elsif session[:jwt_id]
+        Jwt.find_by(id: session[:jwt_id]) || session.delete(:jwt_id) && nil
```

Comment:
If we returned a very simple nil object instead of `nil` here, we could avoid all the uses of the safe navigation operator (`jwt&.jwt_payload` and `jwt&.destroy_stale_states`). Something very basic like this would work:
```
class JwtNilObject
  def jwt_payload
    {}
  end

  def destroy_stale_states; end
end

```
```suggestion
        Jwt.find_by(id: session[:jwt_id]) || session.delete(:jwt_id) && JwtNilObject.new
```

Reply:
That's a nice idea, thanks